### PR TITLE
Request to add monstorak-operator into Openshift-CI framework

### PR DESCRIPTION
**Pull request by aruniiird**
Adding necessary changes to bring 'monstorak/monstorak-operator' into
Openshift-CI system.  Followed the documentation and made changes according to my 
understanding. Please let me know if any further changes required.

All the needed bots are pre-configured according to the documentation.

About 'monstorak-operator'
---------------------------------------
This is a minimalistic operator (inturn uses mixins project),
which helps OCS to bring alerts for the underlying storage.

Signed-off-by: aruniiird <arun.iiird@gmail.com>

### CODE REVIEW

**Comment by droslean** [comment]
@aruniiird Please use the latest [ci-operator-prowgen](https://github.com/openshift/ci-operator-prowgen/tree/master/cmd/ci-operator-prowgen) to generate the jobs.

File: ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+canonical_go_repository: github.com/monstorak/monstorak-operator
+
+base_images:
+  os:
+    cluster: https://github.com/monstorak/monstorak-operator
```
Comment (droslean) [inline ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: ????

File: ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#  inputs:
+#    upgrade-manifest-build:
+#      paths:
+#      - destination_dir: .
+#        source_path: /go/src/github.com/monstorak/monstorak-operator/out/manifests.tar.gz
+#  dockerfile_path: openshift-ci/Dockerfile.registry.build.next
+#  to: monstorak-operator-registry-next
+
+promotion:
+  cluster: https://github.com/monstorak/monstorak-operator
```
Comment (droslean) [inline ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: ???

Comment (droslean) [reply ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: our cluster is `https://api.ci.openshift.org`

Comment (aruniiird) [reply ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: Thanks @droslean , I have changed the cluster value. I was a bit confused with the value, thanks again for clearing it.
With the new PR, I have generated the files with latest 'ci-operator-prowgen'.

Comment (aruniiird) [reply ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: Hi @droslean , I am not sure, why the errors/failures are happening. Regenerated all the files with latest 'ci-operator-prowgen'. Going through the error messages again...

**Comment by droslean** [comment]
> Hi @droslean , I am not sure, why the errors/failures are happening. Regenerated all the files with latest 'ci-operator-prowgen'. Going through the error messages again...

The jobs are generated correctly. Only the jobs your own jobs that you are configuring are failing. We are rehearsing them for you before merging by giving also the chance to debug them.

File: ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+build_root:
+ project_image:
+   dockerfile_path: openshift-ci/Dockerfile.tools
+
+binary_build_commands: make build
+
+tag_specification:
+  name: "4.0"
+  namespace: openshift-storage
```
Comment (droslean) [inline ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: this doesn't exist. This is also why your rehearsals are failing. It should be `ocp`

File: ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        source_path: /go/src/github.com/monstorak/monstorak-operator/out/operator 
+  dockerfile_path: openshift-ci/Dockerfile.deploy
+  to: monstorak
+- from: operator-registry
+  dockerfile_path: openshift-ci/Dockerfile.registry.intermediate
+  to: operator-registry-base
+- from: operator-registry-base
+  dockerfile_path: openshift-ci/Dockerfile.registry.build
+  to: monstorak-operator-registry
+# - from: src
```
Comment (droslean) [inline ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: Please avoid leaving comments in this file. If they are needed, please leave a comment with the reason also.

**Comment by aruniiird** [comment]
> > Hi @droslean , I am not sure, why the errors/failures are happening. Regenerated all the files with latest 'ci-operator-prowgen'. Going through the error messages again...
> 
> The jobs are generated correctly. Only the jobs your own jobs that you are configuring are failing. We are rehearsing them for you before merging by giving also the chance to debug them.

Oh! ok. I gave them as a place holder for now, while I was creating those tests (in parallel). Currently I just want some lint (like golint etc) checks to start while a PR is checked in. I will be adding the tests for a later stage. I will make the targets / labels dummy now (will change them once I have the tests in place)

Thank you @droslean ...

**Comment by aruniiird** [comment]
Hi @stevekuznetsov , @droslean ,
We are trying to bring openshift-ci integration to monstorak operator.  Rectified 'test' and 'lint' issues. Now the error showing is 'Required label not set'. Please help.

/assign @stevekuznetsov

**Comment by aruniiird** [comment]
@stevekuznetsov , @droslean , let me know if something is missing from my end. Trying to get this a closure.
Thanks,
Arun

File: ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      paths:
+      - destination_dir: .
+        source_path: /go/src/github.com/monstorak/monstorak-operator/build/_output/bin/monstorak-operator 
+  dockerfile_path: openshift-ci/Dockerfile.deploy
+  to: monstorak-operator
+
+promotion:
+  cluster: https://api.ci.openshift.org
+  namespace: openshift
+  name: monstorak-0.0.1
```
Comment (droslean) [inline ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: this imagestream doesn't exist. Is this intentional? You will need also to add an imagestream creation as well.

Comment (petr-muller) [reply ci-operator/config/monstorak/monstorak-operator/monstorak-monstorak-operator-master.yaml]: @droslean I think ci-operator creates target ImageStreams if they do not exist on promoting.

https://github.com/openshift/ci-tools/blob/master/pkg/steps/release/promote.go#L83

**Pull request merged**